libraries/AP_AdvancedFailsafe/AP_AdvancedFailsafe.cpp

```diff
@@ -358,13 +360,40 @@ bool AP_AdvancedFailsafe::should_crash_vehicle(void)
         setup_IO_failsafe();
     }
 
-    // should we crash the plane? Only possible with
-    // FS_TERM_ACTTION set to 42
-    if (!_terminate || _terminate_action != 42) {
-        // not terminating
+    // determine if the vehicle should be crashed
+    // only possible if FS_TERM_ACTION is 42 and _terminate is non zero
+    // _terminate may be set via parameters, or a mavlink command
+    if (_terminate_action == TERMINATE && _terminate) {
```

Comment:
I realize this was me getting slightly carried away as there was no functional change here, but the if (!foo || !bar) { false } was making my brain hurt when trying to just validate the entry conditions. I found this a lot more readable with the restructuring here.